src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.cs

```diff
@@ -99,7 +103,7 @@ private protected set
             {
                 Debug.Assert(!IsConfigured, "We should not mutate configured JsonTypeInfo");
                 _serialize = value;
-                HasSerializeHandler = value != null;
+                CanUseSerializeHandler = value != null;
```

Comment:
are we not adding `&& options.CanUseFor...`?

Reply:
`options` can still be mutable at this stage, so this is deferred for the configure stage.